base/array.jl

```diff
@@ -228,6 +228,8 @@ convert{T,n}(::Type{Array{T}}, x::Array{T,n}) = x
 convert{T,n}(::Type{Array{T,n}}, x::Array{T,n}) = x
 convert{T,n,S}(::Type{Array{T}}, x::Array{S,n}) = convert(Array{T,n}, x)
 convert{T,n,S}(::Type{Array{T,n}}, x::Array{S,n}) = copy!(similar(x,T), x)
+convert{T,n,S}(::Type{Array{T}}, X::AbstractArray{S, n}) = copy!(Array(T, size(X)), X)
```

Comment:
> Agreed. I submitted an updated PR.
Thanks. Could you squash the commits? (You can do that using `git rebase -i HEAD~~` and them squashing.)

> On another note, even the convert{T,n} methods could be wrapped into these two methods. But, we would loose the type checking for same types.
Indeed, though it might be clearer written that way... not a big deal IMHO.

> One observation, is that the first pair of convert methods do not make a copy, but the new ones do.

Yes, but I think that's a more general issue which has been discussed somewhere.